Tilemap data generation

julianjriley opened the pull request:
TIlemap Data Generator up and running

- Place BP_TilemapDataGenerator on the center tile of the map
- Expand encapsulationVolume box extent property to fit entire map
- Give a name to the JSON map (won't run if you don't do this)
- Make sure that the Tile Size and Height Factor values are correct
- Click Create Tile Map (will overwrite map with same name)
- Saved tilemap data will be in Content/Data/MapFiles

theshieldboy committed:
test push

WilliamRockwell192 committed:
refactored pathfinding, added skeleton of command system

WilliamRockwell192 committed:
removed old pathfinding code

Shieldboy committed:
Added new UI for deploying units in game

Shieldboy committed:
Separating code from Roster Selection WBP

WilliamRockwell192 committed:
command system implemented and progress on undoing actions

julianjriley committed:
Tilemap Generator Module added as a  private dependency to build

WilliamRockwell192 committed:
refactored player state machine

WilliamRockwell192 committed:
improved undoing movement

Shieldboy committed:
Merge branch 'main' into UnitDeploymentandCommandSystem-WWR,NW

julianjriley committed:
TilemapDataGenerator class implementation

-Fully functional
-Blueprint version in Actors
-Must be given a name to run (help prevent accidental overwriting)
-Data is stored in Data/MapFiles

julianjriley committed:
Merge branch 'main' into TilemapDataGeneration

julianjriley committed:
Changed function name to be more clear

ThomasLarsonBusiness committed:
Fixed Build Error Involving FMOD Event

There was a line missing from a merge conflict involving and FMOD audio event. I simply added that event back in.

Thisisnotdalton commented on Source/TileGenerator/TileGenerator.Build.cs:
```diff
@@ -0,0 +1,33 @@
+            "InputCore", 
+            "EnhancedInput", 
+            "UMG",
+			"TileGenerator"
```
My compiler was giving me warnings about a circular dependency due to the TileGenerator module depending on itself. Removing this line seemed to resolve the warning. How necessary is it to list the TileGenerator as a dependency inside this file?

Shieldboy committed:
Added new trace channel. Work on drag and drop spawning

julianjriley committed:
Merge remote-tracking branch 'origin/Artist_WaterTemple_Blocking' into TilemapDataGeneration

julianjriley committed:
Rectangular tilemap generation

julianjriley committed:
Merge branch 'main' into TilemapDataGeneration

julianjriley committed:
Merge remote-tracking branch 'origin/UnitDeploymentandCommandSystem-WWR,NW' into TilemapDataGeneration

Shieldboy committed:
line trace for hover originating from camera

julianjriley committed:
Tile generation from JSON now works; pathfinding algorithms changed for height inclusion

New tile type: Elevator
Elevator not affected by height differentiation movement costs

All vectors organized using FVector4 instead of FVector. QRSE maps directly to XYZW.
Everything seems fully functional

Candygonemad commented:
<img width="1448" height="663" alt="image_2025-07-24_165348102" src="https://github.com/user-attachments/assets/016a7551-8012-4734-9776-5ad7e3695f21" />

<img width="308" height="102" alt="image" src="https://github.com/user-attachments/assets/75ec8d51-c202-49ec-9a38-3aa15e1b059b" />


Tiles are off from the actual structures. (If when you click play the level is mostly black, leave play mode, go to Window, Levels, and click the pen button in pic 2, then check Initially Loaded and Initially Visible)

<img width="578" height="476" alt="image" src="https://github.com/user-attachments/assets/b8b4c74b-1820-45bc-b358-ede9192e200e" />

Some areas have double tiles that shouldn't like stairs.

<img width="805" height="554" alt="image" src="https://github.com/user-attachments/assets/b05526ca-62ab-4f79-ac87-b44226670057" />

Some parts of the tops of walls have tiles when they shouldn't.

<img width="584" height="505" alt="image" src="https://github.com/user-attachments/assets/8b20238f-340d-4705-9dc4-e25d7d4aa4fc" />

I think it was mentioned, but when creating stairs the tiles should be angles to fit the stairs. If that is a task you have planned for later, that is understandable, and I don't think matters too much currently to the functionality.

Once we get this all worked out, we will want to change the base mesh for the hover mesh.

julianjriley committed:
Tile Generation Changes

- Tile generation now occurs before PIE(Play-In-Editor) mode, allowing for adjustments to tiles pre-runtime
- Changes made to tile rotations preruntime can be stored and then the data can be saved or overwrite the original file

WilliamRockwell192 committed:
added undoing to targeting

WilliamRockwell192 committed:
Merge branch 'main' into UnitDeploymentandCommandSystem-WWR,NW

WilliamRockwell192 committed:
Added unit deployment functionality, cleaned up a variety of things

Also replaced various SetState calls with a playercontroller SetCurrentState. This was because there were multiple instances were a whole new pointer had to be declared locally to access the state. This way, instead of having to cast to playercontroller then cast to playerstate, just cast to playercontroller. quick, easy, and simple.

Had to move and adjust a few systems revolving around spawning and placing units to accommodate for extra units.

Moved NextPlayerTurn out of pawns beginplay and into StartFight in PreBattleOptions

WilliamRockwell192 committed:
fixed crash when attempting to undo when no unit is selected

WilliamRockwell192 committed:
Re-factored unit deployment, add unit repositioning

julianjriley committed:
Tilemap terrain painting tool working and json to data table feature added

julianjriley committed:
Merge branch 'main' into TilemapDataGeneration

julianjriley committed:
Merge remote-tracking branch 'origin/UnitDeploymentandCommandSystem-WWR,NW' into TilemapDataGeneration

julianjriley committed:
Necessary changes from a while ago

julianjriley committed:
Merge remote-tracking branch 'origin/Artist_WaterTemple_Blocking' into TilemapDataGeneration

julianjriley committed:
Merge branch 'main' into TilemapDataGeneration
